lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2930,6 +2955,10 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
   void visitEnumDecl(EnumDecl *ED) {
     checkUnsupportedNestedType(ED);
 
+    // Force creation of the generic signature.
+    (void) ED->getGenericSignature();
+    dumpGenericSignature(Ctx, ED);
+
```

Comment:
I'm curious why we have to do this now.

Reply:
We force certain requests in TypeCheckDeclPrimary just to ensure diagnostics are emitted before we go on to SILGen.